src/shared/install.c

```diff
@@ -2421,11 +2431,11 @@ int unit_file_link(
                 if (r < 0)
                         return install_changes_add(changes, n_changes, r, *file, NULL);
 
-                q = in_search_path(&lp, *file);
-                if (q < 0)
-                        return q;
-                if (q > 0)
-                        continue;
+                r = underneath_search_path(&lp, *file);
+                if (r > 0)
+                        r = -ETXTBSY;
+                if (r < 0)
+                        return install_changes_add(changes, n_changes, r, *file, NULL);
```

Comment:
was the `install_changes_add` change meant to be in the previous commit?

Reply:
I was being lazy and didn't bother to go back after writing the change here. But I'll do it now.